### PR TITLE
Site visibility dialog

### PR DESCRIPTION
**Pull request by vedg**
Added site visibility dialog

updated the image with new changes 

image

### CODE REVIEW

**Review by davidjenni — CHANGES_REQUESTED** [review]
Popping up a modal dialog as part of the extension's activation is too intrusive to users, even here where it is on the vscode.dev path. Please defer showing that warning (even if modal) when the user actually does an up/download.

**Comment by neerajnandwana-msft** [comment]
> Popping up a modal dialog as part of the extension's activation is too intrusive to users, even here where it is on the vscode.dev path. Please defer showing that warning (even if modal) when the user actually does an up/download.

These changes are part of CELA feedback, this requirement is same as what we have discussed around PAC CLI information message.

**Comment by neerajnandwana-msft** [comment]
> Popping up a modal dialog as part of the extension's activation is too intrusive to users, even here where it is on the vscode.dev path. Please defer showing that warning (even if modal) when the user actually does an up/download.

These changes are part of CELA feedback, and it is same as what we have discussed around PAC CLI information message.

File: src/web/client/extension.ts [hunk 1]
```diff
@@ -34,7 +34,8 @@ export function activate(context: vscode.ExtensionContext): void {
             "microsoft-powerapps-portals.webExtension.init",
             async (args) => {
                 _telemetry.sendTelemetryEvent("StartCommand", { 'commandId': 'microsoft-powerapps-portals.webExtension.init' });
-                vscode.window.showInformationMessage(localize("microsoft-powerapps-portals.webExtension.init.message", "Opening VS Code for the web ..."));
+
+                vscode.window.showInformationMessage("Opening VS Code for the web ...");
```
Comment (vedg) [inline src/web/client/extension.ts:38]: will update this in next pr

File: src/web/client/extension.ts [hunk 2]
```diff
@@ -45,6 +46,14 @@ export function activate(context: vscode.ExtensionContext): void {
                         queryParamsMap.set(pair[0], pair[1]);
                     }
                 }
+
+                if (queryParamsMap.get(SITE_VISIBILITY) === PUBLIC) {
+                    const edit: vscode.MessageItem = { isCloseAffordance: true, title: localize("microsoft-powerapps-portals.webExtension.init.sitevisibility.edit","Edit the site") };
```
Comment (ashishchoudhary001) [inline src/web/client/extension.ts:51]: Please confirm if we need: isCloseAffordance

File: webpack.config.js [hunk 3]
```diff
@@ -85,13 +85,6 @@ const webConfig = {
 			test: /\.ts$/,
 			exclude: /node_modules/,
 			use: [{
-                // vscode-nls-dev loader:
```
Comment (ashishchoudhary001) [inline webpack.config.js:88]: Please put the tracking VSCode dev localization bug link. For localization eventually we will have to bring this back.

**Review by ashishchoudhary001 — APPROVED** [review]

File: webpack.config.js (hunk 3 above)
Comment (vedg) [reply webpack.config.js:88]: https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/2896662

File: src/web/client/extension.ts (hunk 2 above)
Comment (vedg) [reply src/web/client/extension.ts:51]: yes this is needed for default close and Esc button behavior

**Pull request merged**